.circleci/config.yml

```diff
@@ -1,11 +1,44 @@
+# following example here: https://dev.to/jonatasbaldin/a-recipe-for-poetry-and-circleci-1bj6
 version: 2
 jobs:
-  build:
-    branches:
-      only:
-        - master
+  build-and-test:
     docker:
-      - image: circleci/python:3.6.5
+      - image: cimg/python:3.7.4
     steps:
       - checkout
-      - run: make test
\ No newline at end of file
+      - restore_cache:
+          keys:
+            - deps-{{ checksum "poetry.lock" }}
+      # Let's install the dependencies
+      - run:
+          name: Install Dependencies
+          command: |
+            poetry install
+      - save_cache:
+          key: deps-{{ checksum "poetry.lock" }}
+          paths:
+            - /home/circleci/.cache/pypoetry/virtualenvs
+      # Another step, run flake8
+      - run:
+          name: Run flake8
+          command: |
+            poetry run flake8 .
+      # Last step, runs our tests ommiting the dependencies path (so we don't take their coverage into account)
+      # And send our coverage somewhere, in this case, coveralls
+      - run:
+          name: Run Pytest
+          command: |
+            poetry run pytest ./tests/*
+
+workflows:
+  version: 2
+  build-and-test-workflow:
+    jobs:
+      - build-and-test:
+          filters:
+            branches:
+              only:
+                - master
+                - /^changyao.*/
```

Comment:
What's the desired end state? Shouldn't "build-and-test" run for all branches / future pull requests?